rtw_excel_report/models/models.py

```diff
@@ -1118,55 +1118,28 @@ def _compute_sale_order_name(self):
+            categ_name = ""
+            prod = line.product_id
+            prod_tmpl_id = line.product_id.product_tmpl_id
+            if prod and prod_tmpl_id:
+                if prod_tmpl_id.config_ok:  
+                    if prod_tmpl_id.categ_id and prod_tmpl_id.categ_id.name:
+                        categ_name = prod_tmpl_id.categ_id.name
+                    elif prod_tmpl_id.product_no:
+                        categ_name = prod_tmpl_id.product_no
+                    elif prod_tmpl_id.name: 
+                        categ_name = prod_tmpl_id.name   
                 else:
-                    # case product is standard Prod + download payment
-                    if line.product_id.product_tmpl_id.seller_ids and line.order_id.partner_id.id:
-                        matching_sup = None  
-                        for sup in line.product_id.product_tmpl_id.seller_ids:
-                            if sup.name.id == line.order_id.partner_id.id:
-                                matching_sup = sup 
-                                break
-                        if matching_sup:
-                            product_code = ("[" + str(matching_sup.product_code) + "]") if matching_sup.product_code else ''
-                            product_name = str(matching_sup.product_name) if matching_sup.product_name else ''
-                            categ_name = product_code + product_name
-                        else:
-                            categ_name =  line.product_id.product_tmpl_id.name
-                    else:
-                        if line.product_id.product_tmpl_id.default_code:
-                            categ_name = "[" +str(line.product_id.product_tmpl_id.default_code) +"]" + line.product_id.product_tmpl_id.name
-                        else:
-                            categ_name =  line.product_id.product_tmpl_id.name
-                                
+                    if line.name:
+                        categ_name = line.name
+                        
             p_type = ""
             if line.p_type:
-                if line.p_type == "special":
-                    p_type = "別注"
-                elif line.p_type == "custom":
-                    p_type = "特注"
+                p_type = "別注" if line.p_type == "special" else "特注" if line.p_type == "custom" else ""
 
-            prod = ""
-            if isinstance(categ_name, bool):
-                categ_name = "" 
-            if categ_name != "" :
-                if p_type !="":
-                    prod += categ_name + "\n" + p_type
-                else:
-                    prod += categ_name
-            else:
-                if p_type != "":
-                    prod += p_type
-                
-            line.sale_order_line_name_excel = str(prod)            
+            detail = categ_name + "\n" + p_type if categ_name and p_type else categ_name if categ_name else p_type if p_type else ""
```

Comment:
chỗ này cũng vậy, viết if else bt, ko dùng tt 3 ngôi.
có thể viết như sau:
detail = ""

if categ_name and p_type:
    detail = categ_name + "\n" + p_type
elif categ_name:
    detail = categ_name
elif p_type:
    detail = p_type